src/clj/game/core/drawing.clj

```diff
@@ -27,45 +28,56 @@
-                     (effect-completed state side eid))
-                   (when (safe-zero? (remaining-draws state side))
-                     (prevent-draw state side))))
-               (when (< draws-after-prevent draws-wanted)
-                 (let [prevented (- draws-wanted draws-after-prevent)]
-                   (system-msg state (other-side side) (str "prevents "
-                                                            (quantify prevented "card")
-                                                            " from being drawn"))))))))
-
+   (trigger-event state side (if (= side :corp) :pre-corp-draw :pre-runner-draw) n)
+   (let [n (+ n (get-in @state [:bonus :draw] 0))
+         draws-wanted n
+         active-player (get-in @state [:active-player])
+         draws-after-prevent (if (and (= side active-player) (get-in @state [active-player :register :max-draw]))
+                               (min n (remaining-draws state side))
+                               n)
+         deck-count (count (get-in @state [side :deck]))]
+     (when (and (= side :corp) (<= deck-count draws-after-prevent))
+       (win-decked state))
+     (when (< draws-after-prevent draws-wanted)
+       (let [prevented (- draws-wanted draws-after-prevent)]
+         (system-msg state (other-side side)
+                     (str "prevents " (quantify prevented "card") " from being drawn"))))
+     (if (or (and (= side active-player)
+                  (get-in @state [side :register :cannot-draw]))
+             (not (pos? draws-after-prevent))
+             (not (pos? deck-count)))
+       (effect-completed state side eid)
+       (let [to-draw (take draws-after-prevent (get-in @state [side :deck]))
+             drawn (mapv #(move state side % :hand) to-draw)
+             drawn-count (count drawn)]
+         (swap! state update-in [side :register :drawn-this-turn] (fnil #(+ % drawn-count) 0))
+         (swap! state update-in [:stats side :gain :card] (fnil + 0) n)
+         (swap! state update :bonus dissoc :draw)
+         (if suppress-event
+           (effect-completed state side eid)
+           (let [draw-event (if (= side :corp) :corp-draw :runner-draw)]
+             (swap! state assoc-in [side :register :currently-drawing] drawn)
+             (queue-event state draw-event {:cards drawn
+                                            :count drawn-count})
+             (wait-for
+               (checkpoint state nil (make-eid state eid) nil)
+               (wait-for (trigger-event-sync state side (make-eid state eid) (if (= side :corp) :post-corp-draw :post-runner-draw) drawn-count)
+                         (let [eid (make-result eid (-> @state side :register :currently-drawing))]
+                           (swap! state update-in [side :register] dissoc :currently-drawing)
+                           (effect-completed state side eid))))))
```

Comment:
I'm trying to think if there are any negative side-effects of potentially triggering `:post-side-draw` events twice. 

In the scenario where an NEH draw is triggered during DBS we'd fire post draw from the NEH draw. We would then clear `:currently-drawing` before returning to the initial call to `draw` where we'd fire post draw events again but this time with 0 cards based to the event. 

Currently we only have 2 cards that care about post draw events. Jinja City Grid clears its fake "once per turn" so that isn't a big deal to be called twice. There is also Find the Truth which reveals the cards drawn which could potentially trigger on reveal events twice; however, I'm not sure the Runner can event create a scenario like the NEH/DBS/PD scenario. At the very least a `req` could be added to Find the Truth to not fire if there aren't any cards in `:currently-drawn`.